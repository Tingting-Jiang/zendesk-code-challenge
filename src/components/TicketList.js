import React from 'react'



const TicketList = ({tickets, setList, setTicket}) => {
    
    // if (loading) {
    //     return <h2>Loading...</h2>;
    // }
    
    // console.log(tickets);
    
    const ViewTicket = (ticket) => {
        // console.log("event is", ticket);
        setTicket(ticket);
        setList(false);
        // console.log("after setting: ", ticket);
        
    }

    return (
        (tickets !== undefined) && (
        
        <ol className="list-group list-group-numbered mb-4">
            {tickets.map(ticketItem => (
              
                <li key={ticketItem.id}
                    
                    className=" card list-group-item list-group-item-action list-group-flush"
                    onClick={() => ViewTicket(ticketItem)}
                    id="list-item">

                        {ticketItem.subject}
                        <span className="float-end">Priority âž¡ {ticketItem.status === null ?
                    "Unknown" : ticketItem.status}</span>

                </li>
            ))}
        </ol>
        )
    
    )
}
export default TicketList;