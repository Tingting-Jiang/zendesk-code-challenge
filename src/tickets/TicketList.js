import React, { useEffect } from 'react'
import { fetchTickets } from './fetchTickets'

const TicketList = ({tickets, setList, setTicket}) => {
    
    // if (loading) {
    //     return <h2>Loading...</h2>;
    // }
    
    // console.log(tickets);
    
    
    const ViewTicket = (ticket) => {
        // console.log("event is", ticket);
        setTicket(ticket);
        setList(false);
        console.log("after setting: ", ticket);
        
    }

    return (
        
        <ul className="list-group mb-4">
            {tickets.map(ticketItem => (
                <li key={ticketItem.id}
                    className="list-group-item"
                onClick={() => ViewTicket(ticketItem)}>
                    {ticketItem.subject}
                    <span className="d-none">
                        {ticketItem.requester_id}
                    </span>
                
                </li>
            ))}
        </ul>
    
    )
}
export default TicketList;