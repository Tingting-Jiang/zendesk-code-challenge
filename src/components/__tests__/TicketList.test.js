import React from 'react';
import TicketList from '../TicketList'
import {shallow, mount, render } from"enzyme";
import testData from "../../data/test.json";


describe("<TicketList/>", () =>{
    it("render TicketList without data", ()=>{
        const wrapper = shallow(<TicketList/>);
        expect(wrapper).toMatchSnapshot();
    })
    
    it("render TicketList with 1 ticket", ()=>{
        const tickets = testData;
        const setList= jest.fn();
        const setTicket = jest.fn();
        const viewTicket = jest.fn();
        const wrapper = shallow(<TicketList tickets={tickets}
                                            setList={setList}
                                            setTicket={setTicket}/>);
        
        expect(wrapper.find('ol').children()).toHaveLength(1);
        expect(wrapper.find('button').text()).toBe('Details');
    })
    
    // it("render TicketList with 1 ticket", ()=>{
    //     const tickets = testData;
    //     const setList= jest.fn();
    //     const setTicket = jest.fn();
    //     const ViewTicket = jest.fn();
    //     const wrapper = shallow(<TicketList tickets={tickets}
    //                                         setList={setList}
    //                                         setTicket={setTicket}/>);
    //
    //     wrapper.find('button').simulate('click');
    //     expect(ViewTicket).toHaveBeenCalled();
    // })
    
    
    
    
    
    
    
    
})