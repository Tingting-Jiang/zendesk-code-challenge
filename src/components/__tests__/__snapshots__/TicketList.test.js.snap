// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TicketList/> render TicketList without data 1`] = `ShallowWrapper {}`;
